### ISSUE TITLE
Should `no-conditional-expect` prevent Promise `catch` in tests?

### ISSUE THREAD

**Issue by manovotny** [jest-community/eslint-plugin-jest#818]
We recently added `no-conditional-expect` to our app, and in trying to eliminate try / catch blocks, discovered that the following is possible. 

```
it('foo', async () => {
    await something
        .then(() => throw 'should have thrown an error')
        .catch((error) => {expect(error).toBe('error')})
})
```

Is that expected or is that a loophole and skirting the intentions of the rule?

**Comment by G-Rath** [comment]
Yeah probably 😅 

tbh I half want to say "no, it only cares about syntax-level conditions" along with an example of how you can easily write a function with a callback, but `valid-expect-in-promise` looks for promises and hasn't had a lot of complaints about false positives, so we might be able to catch this without too much trouble.

I'll see what I can come up with :)
